ts/Series/Map/MapSeries.ts

```diff
@@ -1006,6 +1007,8 @@ class MapSeries extends ScatterSeries {
             if (mapData.type === 'FeatureCollection') {
                 this.mapTitle = mapData.title;
                 mapData = H.geojson(mapData, this.type, this);
+            } else {
+                mapData = isArray(mapData) ? mapData : mapData.features;
```

Comment:
What is the use case here? As I understand it, we either pass a full GeoJSON structure, in case the type must be FeatureCollection, or we pass the results of Highcharts.geojson, in chich case the result is an array. I'm concerned we are introducing more sloppy data structures, which may backfire. 

Also note we are about to [throw TopoJON into the mix](https://github.com/highcharts/highcharts/pull/16533/files#diff-928c6c929e43a646c2dbb842adc8ae6c2012597d40928d41e74f6b085948e2d0R1007-R1010).

Reply:
When updating `series.update({ mapdata: ... })`, the `mapData.type` is being cleaned https://github.com/highcharts/highcharts/blob/d324ceb1af2d64caecd1a6094b04c27c199e0b47/ts/Core/Series/Series.ts#L4133
so the `mapData.type === 'FeatureCollection'` condition if `false`

Not sure how to solve it without interference to the Highcharts core

Reply:
Okay... I need to think about that. Without the type it is not valid GeoJSON, so we need to find some other way.

Reply:
A merge of the current options (in userOptions) and the new options in [here](https://github.com/highcharts/highcharts/pull/16613/files#diff-928c6c929e43a646c2dbb842adc8ae6c2012597d40928d41e74f6b085948e2d0R1291) instead of just setting the new options might resolve this issue.